### PR TITLE
Fix make CFLAGS="-DNDEBUG" failure

### PR DESCRIPTION
**Pull request by awarus**
Redefining NDEBUG in files where assert.h is included to prevent build fail due to -DNDEBUG flag
Fix for #341 issue
Signed-off-by: Ivan Efremov <i350300800e@gmail.com>

### CODE REVIEW

**Comment by jlevon** [comment]
Thanks for this, but I don't think this makes sense as a change we want, thoughts @tmakatos?

**Comment by awarus** [comment]
> Thanks for this, but I don't think this makes sense as a change we want, thoughts @tmakatos?	

I saw the opened issue and comment of @tmakatos. If it's not needed, then we should close the issue or specify in docs this limitation

**Comment by tmakatos** [comment]
> Thanks for this, but I don't think this makes sense as a change we want, thoughts @tmakatos?

While we can't expect the user to use any kind of complication/linking flags they want, I `DNDEBUG` is a very common one so I think it's worth fixing.

**Comment by jlevon** [comment]
The problem is that NDEBUG has a very specific meaning: disable assertions. But we don't - and shouldn't - support doing that. We always build with assertions on. So it doesn't make sense to support it.

And this version - which basically just ignores the requested CFLAGS - doesn't seem right either.

**Comment by awarus** [comment]
> The problem is that NDEBUG has a very specific meaning: disable assertions. But we don't - and shouldn't - support doing that. We always build with assertions on. So it doesn't make sense to support it.
> 
> And this version - which basically just ignores the requested CFLAGS - doesn't seem right either.

Do you mean something like just a sanity check of CFLAGS in Makefile before building the project and some message that DNDEGUB force disabled?

**Comment by jlevon** [comment]
Sure, we could do something like that if you like.

File: Makefile [hunk 1]
```diff
@@ -147,6 +147,9 @@ realclean:
 	rm -rf $(GCOVS)
 
 $(BUILD_DIR)/Makefile:
+ifneq (,$(filter -DNDEBUG,$(CFLAGS)))
+	$(error -DNDEBUG flag is not support)
```
Comment (jlevon) [inline Makefile:151]: nit, this should say "supported"

**Review by jlevon — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (awarus) [reply Makefile:151]: Yeah, sure

**Review by jlevon — APPROVED** [review]

**Pull request merged**